10_rest/nickolee/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/nickolee/pkg/puppy/store"
+
+	puppy "github.com/anz-bank/go-course/10_rest/nickolee/pkg/puppy"
+	rest "github.com/anz-bank/go-course/10_rest/nickolee/pkg/puppy/rest"
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args = os.Args[1:]
+	// this loads your file once parseFlags() is run
+	// these are methods too prepare a variable. Basically telling kingpin how to read the flag from cli
+	fileName = kingpin.Flag("data", "This flag supplies a puppy JSON file name").
```

Comment:
While the input is a filename, kingpin is returning a file struct. This variable should probably be named so that is suggest it's not merely a `string`, eg:
`datafile`